ssh/session/session.go

```diff
@@ -271,6 +272,44 @@ func (s *Session) Register(_ gliderssh.Session) error {
 	return nil
 }
 
+// ConnectionAnnouncement retrieves the connection announcement of the device's namespace.
+// A connection announcement is a custom message provided by the end user that can be printed
+// when a new connection within the namespace is established.
+//
+// Returns the announcement or an error, if any. If no announcement is set, it returns an empty string.
+func (s *Session) ConnectionAnnouncement() (string, error) {
+	device := metadata.RestoreDevice(s.Client.Context())
+	if device == nil {
+		return "", nil
+	}
+
+	res, err := resty.
+		New().
+		R().
+		Get(fmt.Sprintf("http://api:8080/api/namespaces/%s", device.TenantID))
+	if err != nil {
+		return "", err
+	}
+
+	if res.StatusCode() != http.StatusOK {
+		return "", fmt.Errorf("unable to retrieve the namespace's connection announcement. failed with status %d", res.StatusCode())
+	}
+
+	// Response is a temporary type that only needs to handle the announcement
+	type Response struct {
+		Settings struct {
+			Announcement string `json:"connection_announcement"`
+		} `json:"settings"`
+	}
+
+	body := new(Response)
+	if err := json.Unmarshal(res.Body(), body); err != nil {
+		return "", err
+	}
```

Comment:
suggestion: we should set this HTTP logic inside `internalclient`.

Reply:
Initially, I considered creating an internal GET route to specifically retrieve the connection announcement, something like `internal/namespaces/:tenant/connection-announcement`. However, I didn't implement it at that time because I thought it was too much effort.

However, it definitely agree with this. Do you think it's okay to create this route, or is it too excessive?

Reply:
The route I believe is too much, but adding a function to `internalclient` to get namespace information could be very useful.

Reply:
Right, I will implement this and keep you updated.